predicators/envs/pddl_procedural_generation.py

```diff
@@ -618,9 +621,14 @@ def _generate_gripper_problem(
     goal_balls = rng.choice(sorted(list(ball_objects)),
                             size=num_goal_balls,
                             replace=False)
+    possible_goal_rooms = list(range(num_rooms))
     for goal_ball in goal_balls:
+        possible_goal_rooms.remove(initial_ball_rooms[goal_ball])
+        goal_room = rng.choice(possible_goal_rooms, 1)[0]
```

Comment:
does `goal_room = rng.choice(possible_goal_rooms)` work?

Reply:
oh wait yeah it does, nice! - changed